cms/__init__.py

```diff
@@ -60,21 +60,28 @@
 LANG_CPP = "cpp"
 LANG_CS = "cs"
 LANG_PASCAL = "pas"
-LANG_PYTHON = "py"
+LANG_PYTHON2 = "py2"
+LANG_PYTHON3 = "py3"
 LANG_PHP = "php"
 LANG_JAVA = "java"
 
+# Python 3 py_compile outputs to __pycache__/filename.<PYTHON3_COMPILE_NAME>.pyc
+# Using a '*' in this place does not appear to work correctly.
+# Format seems standard as cpython-<MAJOR><MINOR> for cpython.
+PYTHON3_COMPILE_NAME = "cpython-%d%d" % (3, 4)
```

Comment:
I don't know enough about Python's spec, but I'd hazard a guess that
`__pycache__` is cpython specific anyway.

I'm not a fan of the lsdir approach because I doubt we're guaranteed that
python won't produce intermediate files. If the filename convention changes
then maybe we revisit, but that feels less likely than the compiler
producing intermediate or extra files or etc.

Subprocess will also only fork once when the cms file is loaded, lsdir
probably runs a fork underneath and who knows how well that'll play with
cms given it's in the compile step (which is the first step that invokes a
user controlled file, so open to attack vectors and thus sandboxing).
On 10 Jun 2015 7:07 pm, "AdGold" <notifications@github.com> wrote:

> In cms/__init__.py
> <https://github.com/aioc/cms/pull/2#discussion_r32184814>:
>
> >  LANG_PHP = "php"
> >  LANG_JAVA = "java"
> >
> > +# Python 3 py_compile outputs to __pycache__/filename.<PYTHON3_COMPILE_NAME>.pyc
> > +# Using a '*' in this place does not appear to work correctly.
> > +# Format seems standard as cpython-<MAJOR><MINOR> for cpython.
> > +PYTHON3_COMPILE_NAME = "cpython-%d%d" % (3, 4)
>
> Sure, that makes sense, if we're going to that effort though, would it
> make more sense to just do an os.listdir('__pychache__') to get the
> actual file name?
> The advantage would be that it's a) local to the part of the code that
> cares and b) if a non-cpython implementation was used it would also deal
> with that, unlike hardcoding 'cpython' into the name.
> The disadvantage is that it would call os.listdir every time a python3
> program is compiled.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aioc/cms/pull/2/files#r32184814>.
>